python/ray/train/_internal/storage.py

```diff
@@ -463,6 +463,13 @@ def __init__(
             self.storage_path, storage_filesystem
         )
 
+        if not self.storage_filesystem or isinstance(
+            self.storage_filesystem, pyarrow.fs.LocalFileSystem
+        ):
+            self.storage_local_path = self.storage_path
+        else:
+            self.storage_local_path = default_results_dir
```

Comment:
```suggestion
        if isinstance(
            self.storage_filesystem, pyarrow.fs.LocalFileSystem
        ):
            self.storage_local_path = self.storage_path
        else:
            self.storage_local_path = default_results_dir
```

Reply:
otherwise this will set local path to a s3 uri in the case of

`storage_path="s3://...", storage_filesystem=None`

Reply:
In that case, wouldn't pyarrow resolve the filesystem to s3 filesystem anyway? this is checking `self.storage_filesystem` after parsing, not `storage_filesystem`